File: 1346-check-if-n-and-its-double-exist/1346-check-if-n-and-its-double-exist.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var checkIfExist = function(arr) {
    for (let i=0; i < arr.length; i++) {
        const j = arr.indexOf(arr[i] * 2)
        if (j !== -1 && i !== j) return true
    }
    return false
};

File: 1346-check-if-n-and-its-double-exist/1346-check-if-n-and-its-double-exist.py
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        arr.sort()
        for i, n in enumerate(arr):
            l, r = 0, len(arr)-1
            while l <= r:
                m = (l+r) // 2
                if arr[m] == n * 2 and i != m:
                    return True
                elif arr[m] > n * 2:
                    r = m - 1
                else:
                    l = m + 1
        return False